### DISCUSSION TITLE
Light Sensor BH1750 to work using accessory example - solved

### DISCUSSION THREAD

**Discussion by RKF-FKR** [RavenSystem/esp-homekit-devices#2105] [MEPLHAA Scripts]
I'm having no luck getting any readings in HomeKit - always shows a single number.
I followed the example instructions to build the Light Sensor script, so I am not sure where I have gone wrong.

For SDA, SCL I'm using GPIO 22 and 21 on an ESP32
The clock speed is 400khz as per manufacturers data sheet.
The "dt:[0, 3592] is correct as per the manufacturers data sheet. (address 0x30).

I have tested the BH1750 via espeasy and it reads as expected.  

Perhaps HomeKit is not initialising the I2C interface 

Script compiles, and I can add in homekit, however both HomeKit and the terminal log always shows a constant number (like 15146.7).  My script is below.

Any assistance is greatly appreciated

{
	"c": {
		"ic": [
			[22, 21, 400, 1, 1]
		],
		"io": [
			[
				[0], 6
			]
		],
		"l": 13,
		"i": 1,
		"o": 6,
		"f": 1
	},
	"a": [{
		"t": 50,
		"n": 2,
		"dt": [0, 3592]
	}]

**Comment by Rahulsharma0810** [comment]
I am also trying to integrate the BH1750 Lux Sensor. The above code has given me `! Script`.
```
HAA Setup
! Script
Wifi AP HAA-9A5978 Ch6
Start WEB
```

**Comment by RKF-FKR** [comment]
Solved - the issue rested in the declaration of the I2C bus "dt": [0, 35].
Thanks to other members of the forum.
Here is a script with two relays attached.  I2C bus is on GPIO 12 and 13.  They can be triggered by automations relating to lux values.
Relays attached to GPIO 4 and 5 (just happened that my relay board had jumpers next to 4 and 5)
Network monitoring for testing ("o": 6)

{
	"c": {
		"o": 6,
		"ic": [
			[13, 12, 400]
		],
		"io": [
			[
				[5, 4], 2
			]
		]
	},
	"a": [{
		"t": 50,
		"n": 2,
		"dt": [0, 35],
		"es": [{
				"t": 2,
				"0": {
					"r": [
						[5, 0]
					]
				},
				"1": {
					"r": [
						[5, 1]
					]
				}
			},
			{
				"t": 2,
				"0": {
					"r": [
						[4, 0]
					]
				},
				"1": {
					"r": [
						[4, 1]
					]
				}

			}
		]
	}]
}

**Comment by Rahulsharma0810** [comment]
Excellent, I will try that. What tool do you use to capture  "Network monitoring for testing ("o": 6)" ?
